### ISSUE TITLE
Bug: Links are not clearly differentiated

### ISSUE THREAD

**Issue by joelanman** [mdn/yari#5445]
# Summary

Links are not clearly differentiated from other elements (headings, body text). Sometimes they look identical to other elements (headings) sometimes they have a colour variation.

Sometimes links are obvious from their context (say a browse menu) but otherwise the most accessible way to differentiate links from surrounding elements is to use an underline.

**Comment by caugner** [comment]
Hi @joelanman, thank you for taking the time to report this issue. 👋 

Would you be able to add some URLs and a screenshot to illustrate the issue you're reporting? That would be great. 🙂

**Comment by joelanman** [comment]
its a problem on nearly every page, but as an example:

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input

some links are black, some are red, some are grey, none are underlined as far as I can tell. It's just really unclear what are links and what is just text

**Comment by davidluhr** [comment]
Some of the link text within a nested `<code>` element is rendering with `--text-secondary` (light gray) because of a direct descendant selector:

```css
a > code {
  color: var(--code-color);
}
```

This selector doesn't apply to many of the links in `<dl>`/`<dt>`/ `<dd>` content due to the HTML structure:

```html
<a id="attr-checked" href="#attr-checked">
  <b>
    <code>checked</code>
  </b>
</a>
```

The `<b>` element doesn't have any effect, so it could be removed. Alternatively, the CSS selector code simply be `a code { ... }` instead of `a > code { ... }` to select more than direct descendants.

In addition to this, I found anchor link text within a nested `<code>` element does not provide an underline on hover due to overriding specificity.

This hover style loses:
```css
.main-page-content a:not(.button):focus, .main-page-content a:not(.button):hover {
  text-decoration: underline;
}
```

to this selector:
```css
.main-page-content a:not(.button)[aria-current] {
  text-decoration: none;
}
```

**Comment by joelanman** [comment]
to give one specific example:

<img width="395" alt="screenshot: <input> and <textarea> elements that currently have no value. input is grey, textarea is red" src="https://user-images.githubusercontent.com/1132904/156618747-6a081eb8-3e25-4f03-8608-02eb52e12088.png">

the grey 'input' and red 'textarea' do not contrast sufficiently to accessibly differentiate 'textarea' as a link

**Comment by caugner** [comment]
Hi @joelanman, we have addressed this issue in #5416 by underlining links in the content again (see screenshots on the referenced PR), and this is expected to land before the weekend.

Before:
![image](https://user-images.githubusercontent.com/495429/157673759-43cef226-4526-4032-9c90-710bf0346335.png)

After:
![image](https://user-images.githubusercontent.com/495429/157673841-929c5cf3-5b49-4fb3-98dd-2d81d72661f3.png)

**Comment by e3dio** [comment]
This is a very big change to the site. I think it could have used more feedback before merging and deploying. If you look at other reference websites with large amounts of links, they don't underline all the links because the underline makes the text harder to read and distracts from the text. You can easily tell it's a link because all links are colored. I think these things should get more feedback before making major changes like this

![image](https://user-images.githubusercontent.com/85405955/157677676-9e6ac21d-583c-4465-8287-34fc25adf85f.png)

**Comment by caugner** [comment]
@e3dio Links will be underlined again, like they used to be in the old design, see: https://web.archive.org/web/20220222121729/https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input

And I would like to emphasize that we have made this decision after careful consideration of feedback that we got from the community.

**Comment by joelanman** [comment]
@caugner thank you!

**Comment by e3dio** [comment]
Related to this issue: the links within code are colored differently than the links outside code. These code links are hard to read, they are a greyish version of the normal links. I would propose to color those the same color as normal links as pictured below. I am posting in this issue for relevance, you could move to different issue if anyone agrees with this:

# Before

![chrome_40e1bqKa7K](https://user-images.githubusercontent.com/85405955/157692267-1f6e5fab-0341-46cd-a14c-06cf5e16e7f6.png)

# After

![5NnkOu5u60](https://user-images.githubusercontent.com/85405955/157692255-e9cd5487-d990-4a8d-b0ee-afa14c7105fb.png)

**Comment by schalkneethling** [comment]
Hey @joelanman, we are trialing some improvements to the readability and accessibility of MDN Web Docs and we want your input. Please join in the discussion here: https://github.com/mdn/yari/discussions/5715
